pkgs/by-name/de/decasify/package.nix

```diff
@@ -0,0 +1,65 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+
+  # nativeBuildInputs
+  zstd,
+  pkg-config,
+  jq,
+  cargo,
+  rustc,
+  rustPlatform,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "decasify";
+  version = "0.8.0";
+
+  src = fetchurl {
+    url = "https://github.com/alerque/decasify/releases/download/v${finalAttrs.version}/decasify-${finalAttrs.version}.tar.zst";
+    hash = "sha256-HTUAb/yL3H4B/n/Ecd/fDpnTYiqwco/E07sa6pFIIU4=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit (finalAttrs) pname version src;
+    nativeBuildInputs = [ zstd ];
+    # so the cargo fetcher won't try to run the `./configure` script
+    dontConfigure = true;
+    hash = "sha256-bD8MYufI87j//7dIAnCzmp4yoOaT81Zv1i7rjWpjPlc=";
+  };
+
+  nativeBuildInputs = [
+    zstd
+    pkg-config
+    jq
+    cargo
+    rustc
+    rustPlatform.cargoSetupHook
+  ];
+
+  outputs = [
+    "out"
+    "doc"
+    "man"
+    "dev"
+  ];
+
+  enableParallelBuilding = true;
+
```

Comment:
I don't understand why people use `nix-update-script` for simple packages that use standard GitHub tags / releases - how does that makes a difference for the common updating scripts like `nix-update` and `nixpkgs-update`?

Reply:
> > I don't understand why people use nix-update-script for simple packages that use standard GitHub tags / releases - how does that makes a difference for the common updating scripts like nix-update and nixpkgs-update?
> 
> The biggest reason would be that it allows the package to be updated with [./maintainers/scripts/update.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix?rgh-link-date=2024-12-02T20%3A34%3A24Z). It also usually results in more consistent updates from nixpkgs-review/r-ryantm as you're not longer depending on other repositories to update the package with Repology, or nixpkgs-review supporting other attributes you use like `cargoHash`, `pnpmDeps`, etc. (`nix-update` is much better at this)

Evidently `nixpkgs-update` is updating packages based on GitHub releases, and for sure it has no issues with Rust packages that has a `cargoHash`. Perhaps it doesn't support updating hashes of `pnpmDeps`, but that is not an issue for this package. That `update.nix` script is pretty good, but still since `nix-update` is better I'd still not bother adding an `updateScript` for a trivial case like this.

> Edit: Also I'm not really sure why this PR was merged so soon after this approval...probably should've given OP a chance to add this if they wanted to, _especially_ the test

The `passthru.tests.version` attribute is not so bad, but for classic standard packages like this [`versioncheckhook`](https://nixos.org/manual/nixpkgs/stable/#versioncheckhook) is better, as also [claimed by the docs](https://nixos.org/manual/nixpkgs/unstable/#var-passthru-tests-packages), see also a slightly thorough comparison being worked on [here](https://github.com/NixOS/nixpkgs/pull/348025). Due to that, I thought it is not worth it to wait for 1+ more day for CI just for adding these attributes. The maintainer of course is welcome to do any of the above in a separate PR.